recipe/meta.yaml

```diff
@@ -21,19 +21,19 @@ requirements:
     - {{ compiler('c') }}
     - make                               # [not win]
     - cmake
-    - pkg-config                         # [not win]
+    - pkg-config
```

Comment:
are we sure about removing the restriction of pkg-config on Windows? Does it really work there?

Reply:
It doesn't seem to be needed on Windows, but it doesn't hurt either. What do you think @traversaro?

Reply:
Yes, I actually ensured that most Find**.cmake scripts in ignition-cmake worked also without pkg-config as it was not present in the vcpkg CI system back in 2019. However, pkg-config for single-config build is working perfectly fine on Windows, so I think that the less cross-OS difference we have, the better (exactly to avoid subtle differences such as the one fixed by this PR).